### PR TITLE
Basket feature

### PR DESCRIPTION
**Pull request by christopher-timm**

### CODE REVIEW

File: dlf/Configuration/TCA/tx_dlf_actionlog.php [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+        'cruser_id' => 'user_id',
+        'default_sortby' => 'ORDER BY label',
+        'delete'	=> 'deleted',
+        'iconfile'	=> \TYPO3\CMS\Core\Utility\ExtensionManagementUtility::extRelPath('dlf').'res/icons/txdlfreport.png',
+        'rootLevel'	=> 0,
+        'dividers2tabs' => 2,
+        'searchFields' => 'label,name,crdate',
+    ),
+    'interface' => array (
+        'showRecordFieldList' => 'label,name,creation_date',
```
Comment (sebastian-meyer) [inline dlf/Configuration/TCA/tx_dlf_actionlog.php]: There is no field `creation_date`. Do you mean `crdate` as defined above?

File: dlf/Configuration/TCA/tx_dlf_actionlog.php [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+            'config' => array (
+                'type' => 'input',
+                'size' => 30,
+                'max' => 255,
+                'eval' => 'trim',
+            ),
+        )
+    ),
+    'types' => array (
+        '0' => array ('showitem' => '--div--;LLL:EXT:dlf/locallang.xml:tx_dlf_actionlog.tab1, label;;;;1-1-1, name;;;;2-2-2, file_name;;;;2-2-2, creation_date;;;;2-2-2, count_pages;;;;2-2-2'),
```
Comment (sebastian-meyer) [inline dlf/Configuration/TCA/tx_dlf_actionlog.php]: See above: Do you mean `crdate` instead of `creation_date`?

Is it intentional that the field `user_id` is missing from the list?

File: dlf/Configuration/TCA/tx_dlf_basket.php [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+        'doc_ids' => array (
+            'label' => 'LLL:EXT:dlf/locallang.xml:tx_dlf_basket.docIds',
+            'config' => array (
+                'type' => 'input',
+                'size' => 30,
+                'max' => 255,
+                'eval' => 'alphanum_x',
+            ),
+        ),
+        'fe_user_id' => array (
```
Comment (sebastian-meyer) [inline dlf/Configuration/TCA/tx_dlf_basket.php:62]: In line 17 the User ID field was defined as `user_id`, but here it is named `fe_user_id`. Please check for consistency! (Or is this intentional and there are two distinct ID fields?)

File: dlf/Configuration/TCA/tx_dlf_basket.php [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+            'label' => 'LLL:EXT:dlf/locallang.xml:tx_dlf_basket.feUser',
+            'config' => array (
+                'type' => 'input',
+                'eval' => 'int,unique',
+            ),
+        ),
+    ),
+    'types' => array (
+        '0' => array ('showitem' => '--div--;LLL:EXT:dlf/locallang.xml:tx_dlf_basket.tab1, label;;;;1-1-1, session_id;;;;2-2-2, doc_ids;;;;2-2-2, fe_user_id;;;;2-2-2'),
```
Comment (sebastian-meyer) [inline dlf/Configuration/TCA/tx_dlf_basket.php:71]: See last comment. If you rename the `fe_user_id` field you have to change it here as well.

File: dlf/plugins/toc/flexform.xml [hunk 5]
```diff
@@ -54,7 +54,7 @@
 								<allowed>pages</allowed>
 								<size>1</size>
 								<maxitems>1</maxitems>
-								<minitems>1</minitems>
+								<minitems>0</minitems>
```
Comment (sebastian-meyer) [inline dlf/plugins/toc/flexform.xml:87]: Why did you change this? How does the TOC plugin make any sense without a link target?

File: dlf/plugins/toc/locallang.xml [hunk 6]
```diff
@@ -19,12 +19,18 @@
 			<label index="tt_content.pi_flexform.excludeOther">Show only documents from the selected page</label>
 			<label index="tt_content.pi_flexform.targetPid">Target page (with "DLF: Page View" plugin)</label>
 			<label index="tt_content.pi_flexform.templateFile">Template file</label>
+			<label index="tt_content.pi_flexform.basketButton">Show basket button</label>
+			<label index="tt_content.pi_flexform.targetBasket">Target page (with "DLF: Basket" plugin)</label>
```
Comment (sebastian-meyer) [inline dlf/plugins/toc/locallang.xml:23]: You added those options to the flexform of the `pageview` plugin, but this is the localization file of the `toc` plugin. Could you please check this for consistency?

**Review by sebastian-meyer — CHANGES_REQUESTED** [review]
There are some consistency errors in naming the database fields and adding translations to the flexforms.

But most importantly, the basket plugin itself seems to be missing! There are the definition of the database fields and changes to existing files of other plugins, but no class file, template or flexform of the basket plugin itself.

File: dlf/Configuration/TCA/tx_dlf_basket.php (hunk 3 above)
Comment (christopher-timm) [reply dlf/Configuration/TCA/tx_dlf_basket.php:62]: I renamed it to fe_user_id

File: dlf/plugins/toc/locallang.xml (hunk 6 above)
Comment (christopher-timm) [reply dlf/plugins/toc/locallang.xml:23]: The toc plugin has this configuration options too.

File: dlf/plugins/basket/class.tx_dlf_basket.php [hunk 7]
```diff
@@ -0,0 +1,962 @@
...
+
+        exit;
+
+    }
+
+}
+
+if (defined('TYPO3_MODE') && $TYPO3_CONF_VARS[TYPO3_MODE]['XCLASS']['ext/dlf/plugins/basket/class.tx_dlf_basket.php'])	{
+    include_once($TYPO3_CONF_VARS[TYPO3_MODE]['XCLASS']['ext/dlf/plugins/basket/class.tx_dlf_basket.php']);
+}
```
Comment (sebastian-meyer) [inline dlf/plugins/basket/class.tx_dlf_basket.php]: Please remove XCLASS definition as this is no longer supported in TYPO3. (Just delete those 3 lines.)

File: dlf/plugins/navigation/locallang.xml [hunk 8]
```diff
@@ -33,6 +33,7 @@
 			<label index="zoom-in">Zoom In</label>
 			<label index="zoom-out">Zoom Out</label>
 			<label index="zoom-fullscreen">Fullscreen Mode</label>
+			<label index="addBasket">Add to basket</label>
```
Comment (sebastian-meyer) [inline dlf/plugins/navigation/locallang.xml]: Please remove those unused language entries for the navigation plugin. Since there are no code changes to the plugin itself, why would we need those language entries?

File: dlf/plugins/navigation/locallang.xml [hunk 9]
```diff
@@ -56,6 +57,7 @@
 			<label index="zoom-in">Ansicht vergrößern</label>
 			<label index="zoom-out">Ansicht verkleinern</label>
 			<label index="zoom-fullscreen">Vollansicht</label>
+			<label index="addBasket">Zum Warenkorb hinzufügen</label>
```
Comment (sebastian-meyer) [inline dlf/plugins/navigation/locallang.xml]: Please remove those unused language entries for the navigation plugin. Since there are no code changes to the plugin itself, why would we need those language entries?

File: dlf/plugins/toc/flexform.xml (hunk 5 above)
Comment (sebastian-meyer) [reply dlf/plugins/toc/flexform.xml:87]: I am still interested in your intentions behind this change. ;)

**Review by sebastian-meyer — CHANGES_REQUESTED** [review]
Now we are almost there! Just a few little things left. :)

**Comment by christopher-timm** [comment]
Honestly I dont know why I have changed the "minitems". ;)

**Comment by sebastian-meyer** [comment]
Then please revert the minitems-change and we are ready to merge.
